packages/firestore/src/local/indexeddb_mutation_queue.ts

```diff
@@ -103,38 +89,45 @@ export class IndexedDbMutationQueue implements MutationQueue {
   }
 
   start(transaction: PersistenceTransaction): PersistencePromise<void> {
+    let nextBatchId;
+
     return IndexedDbMutationQueue.loadNextBatchIdFromDb(transaction)
-      .next(nextBatchId => {
-        this.nextBatchId = nextBatchId;
+      .next(batchId => {
+        nextBatchId = batchId;
         return mutationQueuesStore(transaction).get(this.userId);
       })
       .next((metadata: DbMutationQueue | null) => {
         if (!metadata) {
           metadata = new DbMutationQueue(
             this.userId,
             BATCHID_UNKNOWN,
-            /*lastStreamToken=*/ ''
+            /*lastStreamToken=*/ '',
+            BATCHID_UNKNOWN
           );
         }
-        this.metadata = metadata;
+
+        // TODO(multitab): This should only run when no other client is
+        // active, as batch IDs may otherwise go back in time in clients that
+        // have already processed mutations.
```

Comment:
I don't really understand this comment.  Since we're in an IndexedDb transaction, no other client can be accessing the queue metadata...  and since you've gotten rid of the in-memory cache, that should be sufficient, right?  So what is this TODO tracking exactly?

Reply:
+1 to not recycling IDs. I thought that was already a goal of this PR.